test/e2e/gardener/shoot/internal/node/node.go

```diff
@@ -44,6 +45,10 @@ const driverName = "foo.driver.example.org"
 
 // VerifyNodeCriticalComponentsBootstrapping tests the node readiness feature (see docs/usage/node-readiness.md).
 func VerifyNodeCriticalComponentsBootstrapping(ctx context.Context, f *framework.ShootFramework) {
+	if v1beta1helper.IsWorkerless(f.Shoot) {
+		return
+	}
```

Comment:
Instead of adding this condition here, maybe skip calling it for workerless shoots?